indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractBatchIndexTask.java

```diff
@@ -583,7 +583,7 @@ public static boolean isGuaranteedRollup(
     return tuningConfig.isForceGuaranteedRollup();
   }
 
-  public static Function<Set<DataSegment>, Set<DataSegment>> compactionStateAnnotateFunction(
+  public static Function<Set<DataSegment>, Set<DataSegment>> prepareCompactionStateAnnotateFunction(
```

Comment:
Nit: Why is this rename needed?
This method also returns a `Function` same as `CompactionState.compactionStateAnnotateFunction`.

Either way, I think the name of these functions should clarify what they do i.e. `addCompactionStateToSegments`

Reply:
Updated to `addCompactionStateToSegments `